py_zipkin/storage.py

```diff
@@ -195,3 +226,57 @@ def set_default_tracer(tracer):
         return _set_contextvars_tracer(tracer)
 
     return _set_thread_local_tracer(tracer)
+
+
+def _Thread_pre_start(self):
+    self._orig_tracer = None
+    if has_default_tracer():
+        self._orig_tracer = get_default_tracer().copy()
+
+
+def _Thread_wrap_run(self, actual_run_fn):
+    # This executes in the new OS thread
+    if self._orig_tracer:
+        # Inject our copied Tracer into our thread-local-storage
+        set_default_tracer(self._orig_tracer)
+    try:
+        actual_run_fn()
+    finally:
+        # I think this is probably a good idea for the same reasons the
+        # parent class deletes __target, __args, and __kwargs
+        if self._orig_tracer:
+            del self._orig_tracer
+
+
+class TracingThread(threading.Thread):
+    """A tracing-aware Thread subclass.
+
+    When run, if the parent thread had a Tracer, this thread gets a copy
+    of its Tracer.  The copy has the properties that stored spans will be
+    available in the parent thread, but the context stack is a deep copy (i.e.
+    contexts pushed or popped in this thread will not affect the parent
+    thread's Tracer's context stack).
+    """
+
+    def start(self):
+        _Thread_pre_start(self)
+        super(TracingThread, self).start()
+
+    def run(self):
+        _Thread_wrap_run(self, super(TracingThread, self).run)
+
+
+def patch_threading():  # pragma: no cover
```

Comment:
can you add a new `instrumentations` folder and put this in a threading.py file? Thanks!

Reply:
Yup!